ludwig/trainers/trainer_lightgbm.py

```diff
@@ -322,12 +322,57 @@ def run_evaluation(
         # Trigger eval end callback after any model weights save for complete checkpoint
         self.callback(lambda c: c.on_eval_end(self, progress_tracker, save_path))
 
-    def _train(
+    def _train_loop(
         self,
         params: Dict[str, Any],
         lgb_train: lgb.Dataset,
         eval_sets: List[lgb.Dataset],
         eval_names: List[str],
+        progress_tracker: ProgressTracker,
+        save_path: str,
+    ) -> lgb.Booster:
+        name_to_metrics_log = {
+            LightGBMTrainer.TRAIN_KEY: progress_tracker.train_metrics,
+            LightGBMTrainer.VALID_KEY: progress_tracker.validation_metrics,
+            LightGBMTrainer.TEST_KEY: progress_tracker.test_metrics,
+        }
+        tables = OrderedDict()
+        output_features = self.model.output_features
+        metrics_names = get_metric_names(output_features)
+        for output_feature_name, output_feature in output_features.items():
+            tables[output_feature_name] = [[output_feature_name] + metrics_names[output_feature_name]]
+        tables[COMBINED] = [[COMBINED, LOSS]]
+        booster = None
+
+        for epoch, steps in enumerate(range(0, self.num_boost_round, self.steps_per_epoch), start=1):
+            progress_tracker.epoch = epoch
+
+            evals_result = {}
+            booster = self.train_step(
+                params, lgb_train, eval_sets, eval_names, booster, self.steps_per_epoch, evals_result
+            )
+
+            progress_tracker.steps = steps + self.steps_per_epoch
+            # log training progress
+            for data_name in eval_names:
+                loss_name = params["metric"][0]
+                loss = evals_result[data_name][loss_name][-1]
+                metrics = {"Survived": {LOSS: loss}, COMBINED: {LOSS: loss}}
+                self.append_metrics(data_name, metrics, name_to_metrics_log[data_name], tables, progress_tracker)
```

Comment:
Are/could we also log these metrics to stdout, similar to how ECD logs intermediate training metrics?

Reply:
Currently, LightGBM is already logging this to stdout. This PR just logs the loss, but eventually we might want to log intermediate metrics as well